jupyterhub/handlers/pages.py

```diff
@@ -442,17 +454,34 @@ class AdminHandler(BaseHandler):
         # get User.col.desc() order objects
         ordered = [getattr(c, o)() for c, o in zip(cols, orders)]
 
-        users = self.db.query(orm.User).outerjoin(orm.Spawner).order_by(*ordered)
+        users = (
+            self.db.query(orm.User)
+            .outerjoin(orm.Spawner)
+            .order_by(*ordered)
+            .limit(per_page)
+            .offset(offset)
+        )
         users = [self._user_from_orm(u) for u in users]
         from itertools import chain
 
         running = []
         for u in users:
             running.extend(s for s in u.spawners.values() if s.active)
 
+        total = self.db.query(orm.User.id).count()
+        pagination = Pagination(
+            url=self.request.uri,
+            total=total,
+            record_name='users',
+            display_msg='Displaying {record_name} <b>{start} - {end}</b>. Total {record_name}: {total}',
+            page=page,
+            per_page=per_page,
+        )
+
         auth_state = await self.current_user.get_auth_state()
         html = self.render_template(
             'admin.html',
+            request=self.request,
```

Comment:
request isn't used in the template, looks like?

Reply:
I think so. Will check it